src/commands/Checkin.ts

```diff
@@ -68,6 +71,23 @@ export default class Checkin extends Command {
     // Get arguments. Get rid of the null types by checking them.
     const publicArgument = interaction.options.getBoolean('public');
     const widescreenArgument = interaction.options.getBoolean('widescreen');
+    const dateArgument = interaction.options.getString('date');
+
+    // Regex to match dates in the format of MM/DD(/YYYY) or MM-DD(-YYYY).
+    const regexp = new RegExp('^(\\d{1,2})(/|-)(\\d{1,2})((/|-)(\\d{2}|\\d{4})){0,1}$', 'g');
+    const dateMatches = regexp.exec(dateArgument!);
+
+    if (dateArgument !== null && dateMatches === null) {
+      await super.respond(interaction, {
+        content: 'Invalid date format. Please use MM/DD or MM-DD format.',
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const month = dateMatches?.[1] ? parseInt(dateMatches[1], 10) : DateTime.now().month;
+    const day = dateMatches?.[3] ? parseInt(dateMatches[3], 10) : DateTime.now().day;
+    const year = dateMatches?.[6] ? parseInt(dateMatches[6], 10) : DateTime.now().year;
```

Comment:
oop lets also check that the date is valid i.e. month is between 1 and 12 and day is between 0 and 31

Reply:
why 0 for day?

ideally backend should just store the number of days/milliseconds since unix epoch, rather than storing each component of the representation separately, then leave date presentation to frontend. this way you can easily prevent 2023-02-29

you might also want to think about time zones